src/cryptonote_core/service_node_list.cpp

```diff
@@ -321,9 +329,201 @@ namespace service_nodes
       LOG_PRINT_L1("Deregistration for service node: " << key);
     }
 
-
     m_rollback_events.push_back(std::unique_ptr<rollback_event>(new rollback_change(block_height, key, iter->second)));
     m_service_nodes_infos.erase(iter);
+
+    return true;
+  }
+
+  static uint64_t get_new_swarm_id(uint64_t seed)
+  {
+    std::mt19937_64 mersenne_twister(seed);
+    return uniform_distribution_portable(mersenne_twister, UINT64_MAX);
```

Comment:
It is just as possible to return an existing swarm_id though. At the same time it is not harmful to put the node in the queue -> they will just be used in the next block. Might be a bigger problem if it returns an existing id (this will result in a large swarm).

Reply:
I think this warrants a revisit. We don't want to assign to a pre-existing swarm, particularly if it's a full swarm and break the max swarm size invariant.

The point in time this function is called is only when all swarms are actually the maximum size so, hitting any swarm would overflow it essentially and break the invariant. `get_new_swarm_id`'s name itself also suggests that this is a new unused swarm id.

Putting it back into the queue is also unintuitive and they wouldn't be used if there's nothing in the block that triggers a swarm update.